DigitalLearningSolutions.Data.Tests/Services/RegistrationServiceTests.cs

```diff
@@ -840,83 +840,82 @@ public void RegisterDelegateByCentre_calls_data_service_to_set_prn()
         }
 
         [Test]
-        public void PromoteDelegateToAdmin_throws_AdminCreationFailedException_if_delegate_has_no_first_name()
+        public void
+            PromoteDelegateToAdmin_throws_active_admin_already_exists_AdminCreationFailedException_if_active_admin_already_exists()
         {
             // Given
-            var delegateUser = UserTestHelper.GetDefaultDelegateUser(firstName: null);
-            var adminRoles = new AdminRoles(true, true, true, true, true, true, true);
-            A.CallTo(() => userDataService.GetDelegateUserById(A<int>._)).Returns(delegateUser);
-
-            // When
-            var result = Assert.Throws<AdminCreationFailedException>(
-                () => registrationService.PromoteDelegateToAdmin(adminRoles, 1, 1)
-            );
-
-            // Then
-            result.Error.Should().Be(AdminCreationError.UnexpectedError);
-        }
-
-        [Test]
-        public void PromoteDelegateToAdmin_throws_AdminCreationFailedException_if_delegate_has_no_email()
-        {
-            // Given
-            var delegateUser = UserTestHelper.GetDefaultDelegateUser(emailAddress: null);
+            var delegateUser = UserTestHelper.GetDefaultDelegateUser();
+            var adminAccount = UserTestHelper.GetDefaultAdminAccount();
             var adminRoles = new AdminRoles(true, true, true, true, true, true, true);
-            A.CallTo(() => userDataService.GetDelegateUserById(A<int>._)).Returns(delegateUser);
+            A.CallTo(() => userDataService.GetDelegateUserById(2)).Returns(delegateUser);
+            A.CallTo(() => userDataService.GetUserIdFromDelegateId(delegateUser.Id)).Returns(1);
+            A.CallTo(() => userDataService.GetAdminAccountsByUserId(1)).Returns(new[] { adminAccount });

```

Comment:
This will probably get changed with the new GetDelegateEntityById or GetUserById changes, but could we specify the IDs used here (either from setting as consts, or taking them from the Default...Account Ids)?